test/Common.FunctionalTests/Utilities/Helpers.cs

```diff
@@ -209,5 +210,29 @@ public static void AllowNoLogs(this IISDeploymentResult deploymentResult)
                 Path.Combine(deploymentResult.DeploymentParameters.PublishedApplicationRootPath, "aspnetcore-debug.log"),
                 "Running test allowed log file to be empty." + Environment.NewLine);
         }
+
+        public static string ReadAllTextFromFile(string filename, ILogger logger)
+        {
+            try
+            {
+                return File.ReadAllText(filename);
+            }
+            catch (Exception ex)
+            {
+                // check if there is a dotnet.exe, iisexpress.exe, or w3wp.exe processes still running.
+                var hostingProcesses = Process.GetProcesses("dotnet").Where(proc =>
```

Comment:
This is a strange condition. `Process.GetProcesses("dotnet")` and then filtering by name.

Reply:
I had that first, but I wanted to get w3wp/iisexpress too. It probably isn't necessary.